scripts/dts/extract_dts_includes.py

```diff
@@ -478,6 +478,24 @@ def main():
     create_aliases(root)
     create_chosen(root)
 
+    # Re-sort instance_id by reg addr
+    # Build a dict of dicts, first level is index by compat
+    # second level is index by reg addr
+    compat_reg_dict = defaultdict(dict)
+    for r in reduced:
+        instance = reduced[r].get('instance_id')
+        if instance and reduced[r]['addr'] is not None:
+            for compat in instance:
+                reg = reduced[r]['addr']
+                compat_reg_dict[compat][reg] = reduced[r]
+
+    # Walk the reg addr in sorted order to re-index 'instance_id'
+    for compat in compat_reg_dict:
+        # only update if we have more than one instance
+        if len(compat_reg_dict[compat]) > 1:
+            for idx, reg_addr in enumerate(sorted(compat_reg_dict[compat])):
+                compat_reg_dict[compat][reg_addr]['instance_id'][compat] = idx
```

Comment:
Maybe it'd be possible to use some `sorted(..., sort=<function>)` thingy here.

Reply:
Seems like overkill for something that will hopefully be short lived.  This is only needed til we can remove the non-DT_INST defines.